tests/src/Integration/Action/UserRoleRemoveTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\UserRoleRemoveTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+use Drupal\Tests\rules\Integration\RulesUserIntegrationTestTrait;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\UserRoleRemove
+ * @group rules_actions
+ */
+class UserRoleRemoveTest extends RulesEntityIntegrationTestBase
+{
+
+  use RulesUserIntegrationTestTrait;
+  /**
+   * The action that is being tested.
+   *
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp()
+  {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->action = $this->actionManager->createInstance('rules_user_role_remove');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary()
+  {
+    $this->assertEquals('Remove roles of particular users', $this->action->summary());
+  }
+
+  /**
+   * Tests removing existing role from user.
+   *
+   * @covers ::execute
+   * @dataProvider rolesProvider
+   * @param $account
+   * @param $user
+   */
+  public function testRemoveExistingRole($account, $user)
+  {
+    // Set-up a mock user with role 'editor'.
+    $editor = $this->getMockedUserRole('editor');
+
+    // Test removing one role.
+    $this->action
+      ->setContextValue('user', $account)
+      ->setContextValue('roles', [$editor])
+      ->execute();
+
+    $this->assertEquals($this->action->autoSaveContext(), $user, 'Action returns nothing for auto saving since the user has been saved already.');
+  }
+
+  /**
+   * @return array
```

Comment:
Please add a short description "Provides mocked user objects for the testRemoveExistingRole test." or somtehing like that.